pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -249,3 +250,27 @@ def on_evaluation_epoch_end(self) -> None:
     def teardown(self) -> None:
         self._results.cpu()
         self.epoch_loop.teardown()
+
+    def _should_reload_val_dl_epoch(self) -> bool:
+        """Check if val dataloader should be reloaded in the current epoch."""
+        current_epoch = self.epoch_loop.batch_progress.current.completed
```

Comment:
```suggestion
        current_epoch = self.trainer.current_epoch
```

Reply:
This was actually self.epoch_progress.current.completed from the fit_loop.

Reply:
Yes. This is one of my question.  I noticed that original implementation is to initialize current_epoch with self.epoch_progress.current.completed from the fit_loop, and since now, we want to give users more granular control, by introducing `reload_train_dataloader_every_n_epochs` and `reload_val_dataloader_every_n_epochs `.Therefore, considering this is evaluation_loop, does it still make sense to get current epoch from fit_loop?